packages/gatsby/src/redux/actions.js

```diff
@@ -86,6 +86,7 @@ const pascalCase = _.flow(
 )
 const hasWarnedForPageComponentInvalidContext = new Set()
 const hasWarnedForPageComponentInvalidCasing = new Set()
+const pageComponentCache = new Map()
```

Comment:
Want to replace this with a WeakMap? WeakMaps can be cleaned up by GC so great for caches

Reply:
This is a great idea! We should also see if this approach improves performance in other areas where we're using Map (where it makes sense to use a WeakMap!).

Reply:
Unfortunately according to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap#Description) primitive values can't be as keys / values for weak maps, only object references.

Reply:
This map won't ever be huge (as this just map template paths), so I don't worry about memory/gc here